### ISSUE TITLE
Invalid nested PEP 585 syntax with postponed evaluation

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pylance-release#953]
<h3> Environment data </h3>

-   Language Server version: 2021.2.2
-   OS and version: darwin x64
-   Python version (and distribution if applicable, e.g. Anaconda): **3.8**
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: strict

<h3> Expected behaviour </h3>
PEP 585 syntax in type aliases is invalid in 3.7 / 3.8 even when nested inside Union / Tuple / Dict.

<h3> Actual behaviour </h3>
No errors

<h3> Logs </h3>
--

<h3> Code Snippet / Additional information </h3>

```python
from __future__ import annotations
from typing import Union

Alias1 = Union[list[str], None]
Alias2 = Union[Union[list[int], int]]
Alias3 = Tuple[list[int]]
Alias4 = Dict[str, list[int]]
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.2.3, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202123-17-february-2021
